packages/algorand-js/src/interpreter/opcode-list.ts

```diff
@@ -151,3 +157,319 @@ export class Intc extends Op {
     stack.push(intc);
   }
 }
+
+// err opcode : Error. Panic immediately.
+export class Err extends Op {
+  execute (stack: TEALStack): void {
+    throw new TealError(ERRORS.TEAL.TEAL_ENCOUNTERED_ERR);
+  }
+}
+
+// SHA256 hash of value X, yields [32]byte
+export class Sha256 extends Op {
+  execute (stack: TEALStack): void {
+    const hash = sha256.create();
+    const val = stack.pop() as Message;
```

Comment:
```suggestion
    const val = stack.pop();
```
I think you get a type error for `undefined` for `val`. I have updated the stack definition, so this will be fine. Sorry.

Reply:
Also, it should be changed to
```
const val = this.assertBytes(stack.pop());
```

Reply:
this code doesn't throw type error, if you see sha256.create(), the hash.update() function takes `Message` type as a input.
we can add a check before , so we don;t take value other than Uint8array